\graphicspath{{chapters/06/images/}}
\chapter{Gene set enrichment}

\section{Introduction}

	\subsection{Functional groups characterized by gene expression change}
	Functional groups characterized by gene expression change aer:

	\begin{multicols}{2}
		\begin{itemize}
			\item Gene sets: the set is scored depending on the expression level of its member genes.
			\item Network: modules satisfying some joint gene expression and topology requirement are identified.
			\item Pathways: they are scored exploiting gene expression and topology.
		\end{itemize}
	\end{multicols}

	\subsection{Gene-set enrichment analysis}
	Gene-set enrichment analysis is the breakdown of cellular functions into gene sets.
	Every set of gene is associated to a specific cellular:

	\begin{multicols}{4}
		\begin{itemize}
			\item Function.
			\item Process.
			\item Component.
			\item Pathway.
		\end{itemize}
	\end{multicols}

	Microarray or RNA-seq data can be related to gene sets in order to mine its functional meaning, to find which gene sets summarize at best gene expression patterns.

	\subsection{Ontology}
	Ontology normally represents knowledge as a set of concepts within a domain and the relationships among those concepts.
	It can be used to reason about the entities within that domain and may be used to describe the domain.

	\subsection{Controlled vocabulary}
	A controlled vocabulary provides a way to organize knowledge for subsequent retrieval but does not allow reasoning about the entities.

	\subsubsection{Ontology and controlled vocabulary uses}
	Ontologies and controlled vocabularies are heavily used in biological databases as they allow the organization of data within a database, providing a meaningful link between database structure and search queries.


\section{Gene ontology}
A gene ontology is a way to capture biological knowledge for individual gene products in a written and computable form.

	\subsection{Concepts hierarchy}
	A set of concepts and their relationships to each other is arranged as a hierarchy.

		\subsubsection{Molecular function}
		The molecular function describes activities that happen at a molecular level like catalytic or binging activity.
		This category includes the activities rather than the entities that are involved in an action and do not specify where, when or in which context the actions happen.
		Molecular functions can be executed by single gene products or complexes of gene products.
		Some examples are:

		\begin{multicols}{4}
			\begin{itemize}
				\item Catalytic activity.
				\item Transport activity.
				\item Binding.
				\item Toll receptor binding.
			\end{itemize}
		\end{multicols}

		\subsubsection{Biological process}
		A biological process is a series of events resulting from multiple ordered groups of molecular functions.
		A biological process is different from a pathway, as gene ontology does not report the dynamics or the dependences that are required to describe a pathway.
		Some examples are:

		\begin{multicols}{4}
			\begin{itemize}
				\item Cellular physiological processes.
				\item signal transduction.
				\item Metabolic process of pyrimidine.
				\item Glucose transport.
			\end{itemize}
		\end{multicols}

		\subsubsection{Cellular component}
		A cellular component is linked to a component of a cell with the condition that is part of a larger object.
		Some examples are:

		\begin{multicols}{4}
			\begin{itemize}
				\item Ribosome.
				\item Nucleus.
				\item Neuron parts.
				\item Internal nuclear membrane.
			\end{itemize}
		\end{multicols}

	\subsection{Ontology structure}
	An ontology is structured as an acyclic graph where terms can have more than one parent.
	Terms are linked by directed relationships like:

	\begin{multicols}{2}
		\begin{itemize}
			\item Is part of.
			\item Regulates.
		\end{itemize}
	\end{multicols}

	\subsection{The gene ontology project}
	Originally in the gene ontology or GO project the hierarchies were completely independent, without links between them.
	From $2009$ biological processes and molecular functions are linked, as biological processes are ordered assemblies of molecular functions.
	GO is required as there are inconsistencies in the human language: different concepts can have the same name.
	Furthermore it enables to interpret quickly large datasets.
	The aim of the GO project is to:

	\begin{multicols}{3}
		\begin{itemize}
			\item Compile ontologies.
			\item Annotate gene products using ontology terms.
			\item Provide a public resource of data and tools for annotation.
		\end{itemize}
	\end{multicols}

	A gene ontology annotation is a statement that a gene product has a particular molecular function, determined by a particular method and described in a reference.

\section{Gene sets}

	\subsection{Sources and types}
	Other than gene ontology there are other sources and types of gene-sets:

	\begin{multicols}{2}
		\begin{itemize}
			\item Pathways (KEGG).
			\item Protein families and domains (PFAM).
			\item Predicted target of regulators like mRNA and transcription factors (MSigDB-c3).
			\item Protein-protein interaction modules.
			\item Gene expression:

				\begin{itemize}
					\item Up and down regulation after treatment or in relation to disease (MSigDB-c2).
					\item Co-expression across many conditions (MSigDB-c4).
				\end{itemize}

			\item Genotype-phenotype association (DiseaseHub).
			\item Genomic position (MSigDB-c1).
		\end{itemize}
	\end{multicols}

		\subsubsection{Main resources}
		The main resources for these type of data are:

		\begin{multicols}{2}
			\begin{itemize}
				\item Bioconductor.
				\item DiseaseHub.
				\item MSigDB.
				\item PathwayCommons.
				\item WhichGenes.
			\end{itemize}
		\end{multicols}

	\subsection{Differences between pathways and processes}
	From a biological perspective the difference between pathways and processes is philosophical.
	It is still worth speculating in a bioinformatics perspective because a gene is annotated for a GO biological process if the curators deem it significantly contributes to the process according to a number of evidences.
	Pathway include the wiring of genes and gene products, hence they rely on a more intensive curation process.
	Some pathways include large ubiquitous actors such as the proteasome that may confound enrichment analysis, whereas the are usually absent from GO processes.

	\subsection{Enrichment test}
	An enrichment test combines a gene expression table with gene-set databases to build an enrichment table.
	In the enrichment table each gene set is associated with a p-value, that gives use the probability that the differentially expressed genes are part of the gene set.
	This is a two class design: genes can be ranked according to different statistics like fold change, log ratio or t-test and a selection by threshold can be performed.
	Also an expression matrix can be obtained with multiple condition employing for example ANOVA.
	A gene set can both overlap with significant gene and background genes.
	To test whether this overlap is significant it must be compared with random sampling of captured genes.

		\subsubsection{Fisher's exact test}
		Fisher's exact test calculates the exact probability of the table of observed cell frequencies in a table given that:

		\begin{multicols}{2}
			\begin{itemize}
				\item THe null hypothesis of independence is true.
				\item The marginal totals of the observed table are fixed.
			\end{itemize}
		\end{multicols}

		It does not require to perform random sampling as it is based on the theoretical null-hypothesis distribution: the hypergeometric distribution.
		In particular it gives the probability that the overlap between significant genes and gene-set is greater than the one expected by random sampling the captured genes.
		Fisher's table is built such that:

		\begin{table}[H]
			\centering
			\begin{tabular}{ccccc}
				\multirow{4}{*}{\rotatebox[origin=c]{90}{\tiny{Up regulated genes}}}& \multicolumn{4}{c}{Gene Set}\\
				 & & No & Yes & \\
				 \cline{3-4}
				 &Yes & \multicolumn{1}{|c|}{$a$} & \multicolumn{1}{|c|}{$b$} & $a+b$\\
				 \cline{3-4}
				 &No & \multicolumn{1}{|c|}{$c$} & \multicolumn{1}{|c|}{$d$} & $c+d$\\
				 \cline{3-4}
				 & & $a+c$ & $b+d$ & \\
			\end{tabular}
			\caption{Fisher's exact test table}
			\ref{tab:fisher}
		\end{table}

		Considering the table \ref{tab:fisher}:

		\begin{multicols}{2}
			\begin{itemize}
				\item $a$ depicts the significant gene not in overlap.
				\item $b$ the significant ones in overlap.
				\item $c$ the background genes not in overlap.
				\item $d$ the background gene in overlap.
			\end{itemize}
		\end{multicols}

		Then the exact probability of the table is:

		$$\frac{(a+b)!\cdot(c+d)!\cdot(a+c)!\cdot(b+d)!}{n!\cdot a!\cdot b!\cdot c!\cdot d!}$$

		The p-value is calculated by summing all probabilities less than or equal to the probability of the observed table.
		Fisher's exact test can be used to evaluate the overlaps between gene-sets from databases.
		It is usually employed in threshold-dependent scenario and suffers from some limitations.

	\subsection{Whole-distribution - GSEA enrichment}
	Whole distribution methods like GSEA, gene set enrichment analysis, have been shown to be more stable and statistically powerful.
	Instead of excluding genes they are ranked according to a measure.
	This type of test can be used to compare $2$ classes of genes and they can be correlated to a phenotype with a Pearson correlation test.

		\subsubsection{Process}
		First the enrichment score is calculated: genes are ranked according to a specific measure like fold change.
		Then the chance of observing a gene in a position towards the upregulated or downregulated region is computed.
		Based on the position in which genes are found, it can be established whether the event is random or not.
		Every gene present in a gene list gives a positive contribution, while every absent one a negative one.
		So, an high enrichment score means high local enrichment.
		After this the enrichment score distribution is generated for the null hypothesis using permutation and the empirical p-value is computed observing how many random data have enrichment score greater than the read ones.
		Then the FDR correction of the p-value is computed.

		\subsubsection{Permutations}
		Permutations setting have important implication and can be used only when biological replicates are very similar within classes or are well separated.
		When biological replicates tend to be dissimilar or stratified according to hidden experimental factors other whole-distribution enrichment methods need to be used.

		\subsubsection{Other uses of GSEA}
		The GSEA tool allow to perform more general types of analysis specifying desings:

		\begin{multicols}{2}
			\begin{itemize}
				\item .GMT data format contains GO ID, description and information of the genes contained in the GO term.
				\item The gene expression table contains CDM or RPKM values for each gene.
				\item The expression phenotype file .cls determine which sample belongs to which class.
			\end{itemize}
		\end{multicols}

	\subsection{Gene set filter}
	Gene set for enrichment analysis are usually filtered by size.
	Large gene-sets are undesired if they are derived from gene ontology or other functional resources as the usually correspond to uninformative concepts.
	Small gene-sets are undesired as their statistics are noisy and may decrease the FDR of other sets.

	\subsection{Redundancy problem}
	There are many redundant gene-sets.
	For example gene ontology has a very large number of gene-sets, often with slight differences.
	Moreover different pathway databases have different but overlapping definitions of pathways.
	Globally it is useful to grasp the overlap relations between enriched gene-sets.
	To do so a visualization framework beyond the enrichment table is needed.
	The redundancy problem can be handled by correcting for inter-redundancy and prioritizing the most enriched gene-sets or by visualizing gene-set overlap as a network with the EnrichMap tool.
	Considering up and down regulation a map can be built such that:

	\begin{multicols}{2}
		\begin{itemize}
			\item Each node is a gene set.
			\item The colour represent differentially expressed genes.
			\item The size of the node is the number of genes.
			\item The edge thickness the overlap or similarity degree.
		\end{itemize}
	\end{multicols}

	The network can be then clustered based on similarity values in order to identify redundancy sets.
